pkg/distributor/push_test.go

```diff
@@ -806,105 +807,128 @@ func TestHandler_ToHTTPStatus(t *testing.T) {
 			expectedErrorMsg:   ingestionRateLimitedErr.Error(),
 		},
-		{
-			name:               "a DoNotLogError of an ingestionRateLimitedError gets translated into an HTTP 429",
+		"a DoNotLogError of an ingestionRateLimitedError gets translated into an HTTP 429": {
 			err:                log.DoNotLogError{Err: ingestionRateLimitedErr},
 			expectedHTTPStatus: http.StatusTooManyRequests,
 			expectedErrorMsg:   ingestionRateLimitedErr.Error(),
 		},
-		{
-			name:                        "a requestRateLimitedError with serviceOverloadErrorEnabled gets translated into an HTTP 529",
+		"a requestRateLimitedError with serviceOverloadErrorEnabled gets translated into an HTTP 529": {
 			err:                         requestRateLimitedErr,
 			serviceOverloadErrorEnabled: true,
 			expectedHTTPStatus:          StatusServiceOverloaded,
 			expectedErrorMsg:            requestRateLimitedErr.Error(),
 		},
-		{
-			name:                        "a DoNotLogError of a requestRateLimitedError with serviceOverloadErrorEnabled gets translated into an HTTP 529",
+		"a DoNotLogError of a requestRateLimitedError with serviceOverloadErrorEnabled gets translated into an HTTP 529": {
 			err:                         log.DoNotLogError{Err: requestRateLimitedErr},
 			serviceOverloadErrorEnabled: true,
 			expectedHTTPStatus:          StatusServiceOverloaded,
 			expectedErrorMsg:            requestRateLimitedErr.Error(),
 		},
-		{
-			name:                        "a requestRateLimitedError without serviceOverloadErrorEnabled gets translated into an HTTP 429",
+		"a requestRateLimitedError without serviceOverloadErrorEnabled gets translated into an HTTP 429": {
 			err:                         requestRateLimitedErr,
 			serviceOverloadErrorEnabled: false,
 			expectedHTTPStatus:          http.StatusTooManyRequests,
 			expectedErrorMsg:            requestRateLimitedErr.Error(),
 		},
-		{
-			name:                        "a DoNotLogError of a requestRateLimitedError without serviceOverloadErrorEnabled gets translated into an HTTP 429",
+		"a DoNotLogError of a requestRateLimitedError without serviceOverloadErrorEnabled gets translated into an HTTP 429": {
 			err:                         log.DoNotLogError{Err: requestRateLimitedErr},
 			serviceOverloadErrorEnabled: false,
 			expectedHTTPStatus:          http.StatusTooManyRequests,
 			expectedErrorMsg:            requestRateLimitedErr.Error(),
 		},
 	}
 
+	// We enrich testCases with the cases corresponding to ingesterPushErrors
+	// with all possible error causes, and ensure that all those errors, except
+	// the ones with mimirpb.BAD_DATA and mimirpb.TSDB_UNAVAILABLE causes will
+	// be translated into an HTTP status 500, while the former will be translated
+	// an HTTP 400 and an HTTP 503 respectively.
+	// Additionally, we enrich testCases with the same ingeseterPushError, but
+	// wrapped with log.DoNotLogError.
+	ingesterPushErrorCauses := map[string]mimirpb.ErrorCause{
```

Comment:
Is this temporary while we're switching to this new way of handling errors? Defining a list of test cases and then modifying the test cases is a bit hard to follow and understand exactly what is being tested.

Reply:
You are right, I made it look more complicated than it should. The test cases I added here are not temporary, they are valid cases that should be tested. I have now added them to the list of tests explicitly.